merlion/models/automl/autoets.py

```diff
@@ -7,21 +7,83 @@
+from statsmodels.tsa.exponential_smoothing.ets import ETSModel
 from merlion.models.forecast.ets import ETS
-from merlion.models.automl.seasonality import SeasonalityConfig, SeasonalityLayer
+from merlion.models.automl.seasonality import PeriodicityStrategy, SeasonalityConfig, SeasonalityLayer
+from merlion.utils import TimeSeries, UnivariateTimeSeries
 
+logger = logging.getLogger(__name__)
 
 class AutoETSConfig(SeasonalityConfig):
     """
-    Config class for ETS with automatic seasonality detection.
+    Configuration class for `AutoETS`. Act as a wrapper around a `ETS` model, which automatically detects
+    the seasonal_periods, error, trend, damped_trend and seasonal.
     """
 
-    def __init__(self, model: Union[ETS, dict] = None, **kwargs):
+    # This is adapted from ets.R from forecast package
+    def __init__(
+        self, model: Union[ETS, dict] = None,
+        auto_seasonality: bool = True,
+        auto_error: bool = False,
+        auto_trend: bool = False,
+        auto_seasonal: bool = False,
+        auto_damped: bool = False,
+        seasonal_periods: int = 1,
+        pred_interval_strategy: str = "exact",
+        periodicity_strategy: PeriodicityStrategy = PeriodicityStrategy.ACF,
+        information_criterion: str = "aic",
+        additive_only: bool = False,
+        allow_multiplicative_trend: bool = False,
+        restrict: bool = True,
+        **kwargs
+    ):
+        """
+        :param auto_seasonality: Whether to automatically detect the seasonality.
+        :param auto_error: Whether to automatically detect the error components.
+        :param auto_trend: Whether to automatically detect the trend components.
+        :param auto_seasonal: Whether to automatically detect the seasonal components.
+        :param auto_damped: Whether to automatically detect the damped trend components.
+        :param seasonal_periods: The length of the seasonality cycle. ``None`` by default.
+        :param pred_interval_strategy: Strategy to compucate prediction intervals. "exact" or "simulated".
+        Note that "simulated" setting supports more variants of ETS model.
+        :param periodicity_strategy: Periodicity Detection Strategy.
+        :param information_criterion: informationc_criterion to select the best model. It can be "aic",
+        "bic", or "aicc".
+        :param additive_only: If True, the search space will only consider additive models.
+        :param allow_multiplicative_trend: If True, models with multiplicative trend are allowed in the search
+        space.
+        :param restrict: If True, the models with infinite variance will not be allowed in the search space.
+        """
         model = dict(name="ETS") if model is None else model
-        super().__init__(model=model, **kwargs)
+        super().__init__(model=model, periodicity_strategy=periodicity_strategy, **kwargs)
+        self.seasonal_periods = seasonal_periods
+        self.pred_interval_strategy = pred_interval_strategy
```

Comment:
`seasonal_periods` and `pred_interval_strategy` should reference `self.model.seasonal_periods` and `self.model.pred_interval_strategy`, since `self.model` is an instance of `ETS`. These parameters shouldn't be duplicated here. Also note that you should be able to pass these parameters to the `AutoETSConfig`, and they should be propagated down to the underlying model automatically.

Reply:
I have fixed it